marlowe-dashboard-client/src/Contract/State.purs

```diff
@@ -43,17 +45,18 @@ emptyMetadata =
   }
 
 defaultState :: State
-defaultState = mkInitialState zero Close emptyMetadata
+defaultState = mkInitialState zero Close emptyMetadata mempty
 
-mkInitialState :: Slot -> Contract -> MetaData -> State
-mkInitialState slot contract metadata =
+mkInitialState :: Slot -> Contract -> MetaData -> Map String String -> State
+mkInitialState slot contract metadata roleWallets =
   { tab: Tasks
   , executionState: initExecution slot contract
   , side: Overview
   , confirmation: Nothing
   , contractId: Nothing
   , step: 0
   , metadata
+  , roleWallets
```

Comment:
I'm not sure about this. Are you thinking you want to map the role name (tokenName) to a wallet nickname? Because I don't think that's going to be possible. The contract presumably won't know which wallet has which role (given that in principle people could sell their role tokens to other people after the contract starts). It should be fine just to show the tokenName though (at least until we think of something better).

Reply:
I'm trying to implement this part of the design 
![Screen Shot 2021-03-12 at 10 47 02](https://user-images.githubusercontent.com/2634059/110948602-500aa880-8320-11eb-96c1-b84d7d29e760.png)

Which is "Role name (role nickname)". To know the different roles I'll use the `getParties` from the Contract, but to possible assign them names, I'll use this information. If I have the nickname I'll display the parentheses, and if I don't I'll just display the role name.

I think this is only going to work for the person setting up the initial contract, and perhaps we can make a participant know it's role.... I still have my doubts on this :/